widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -732,14 +732,18 @@ private void appendMessageItem(List<ChatItem> currentChatItems, ChatMessage mess
             changeLastOperatorMessages(currentChatItems, message);
             appendMessagesNotSeen();
             if (message.getAttachment() instanceof SingleChoiceAttachment) {
-                SingleChoiceAttachment attachment =
-                        ((SingleChoiceAttachment) message.getAttachment());
+                SingleChoiceAttachment attachment = ((SingleChoiceAttachment) message.getAttachment());
                 emitViewState(
                         chatState.chatInputModeChanged(
                                 attachment.getOptions() != null ?
                                         ChatInputMode.SINGLE_CHOICE_CARD :
                                         ChatInputMode.ENABLED
-                        ));
+                        )
+                );
+            } else if (chatState.chatInputMode == ChatInputMode.SINGLE_CHOICE_CARD) {
+                emitViewState(
+                        chatState.chatInputModeChanged(ChatInputMode.ENABLED)
+                );
```

Comment:
method getting too long and complex

Reply:
will need to rewrite incoming message logic at some point, atm. is fine